content/api/commands/attachfile.md

```diff
@@ -0,0 +1,239 @@
+  contents: 'path/to/file.json',
+  fileName: 'users.json',
+  lastModified: new Date('Feb 18 1989').valueOf()
+})
+```
+
+### Dropping a file on the document
+
+```javascript
+cy.document().attachFile('file.json', { action: 'drag-n-drop' })
+```
+
+## Notes
+
+### Existence
+
+#### Default file existence assertion
+
+Whenever resolving a file path, `.attachFile()` asserts that the file exists and will fail if it does not exist. It will retry reading the file if it does not initially exist until the file exists or the command times out.
+
+```javascript
+// will fail after the defaultCommandTimeout is reached
+cy.get('input[type=file]').attachFile('does-not-exist.yaml')
+```
+
+### Actionability
+
+#### The element must first reach actionability
+
+`.attachFile()` is an "action command" that follows all the rules
+of [Actionability](/guides/core-concepts/interacting-with-elements).
+
+## Rules
+
+### Requirements [<Icon name="question-circle"/>](/guides/core-concepts/introduction-to-cypress#Chains-of-Commands)
+
+<List><li>`.attachFile()` requires being chained off a command that yields DOM
+element(s). With the `input` action (default), it further requires a single `input` element with `type="file"`, or a `label` element attached to one.</li><li>If given a path, `.attachFile()` requires the file must exist.</li><li>If given an alias, `.attachFile()` requires that the subject of the alias must not be `null` or `undefined`.</li></List>
+
+### Assertions [<Icon name="question-circle"/>](/guides/core-concepts/introduction-to-cypress#Assertions)
+
+<List><li>`.attachFile()` will automatically wait for the element to reach an
+[actionable state](/guides/core-concepts/interacting-with-elements).</li></List>
+
+### Timeouts [<Icon name="question-circle"/>](/guides/core-concepts/introduction-to-cypress#Timeouts)
+
+<List><li>`.attachFile()` can time out waiting for the element to reach an
+[actionable state](/guides/core-concepts/interacting-with-elements).</li><li>
+`.attachFile()` can time out waiting for a file to exist on disk or for an
+alias to resolve.</li></List>
+
+## Command Log
+
+**_Attach file to input_**
+
+```javascript
+cy.get('.file-input').attachFile(Buffer.from('Hello world'))
+```
+
+The commands above will display in the Command Log as:
+
+<DocsImage src="/img/api/attachfile/attach-file-during-test.png" alt="Command log for attachFile" ></DocsImage>
+
+When clicking on `attachFile` within the command log, the console outputs the
+following:
+
+<DocsImage src="/img/api/attachfile/attach-file-in-console.png" alt="console.log for attachFile" ></DocsImage>
+
+## History
+
+| Version                                       | Changes                                                  |
+| --------------------------------------------- | -------------------------------------------------------- |
+| [10.0.0](/guides/references/changelog#10.0.0) | `.attachFile()` command added                            |
+
+### Community Recognition
```

Comment:
@jennifer-shehane , @cowboy , @jaffrepaul - Calling this section out as it's newly added since the last round of reviews. Just cleaning up some loose ends.

We don't really have a standard for this, but I borrowed the format from https://github.com/cypress-io/cypress/blob/develop/npm/cypress-schematic/README.md#community-recognition.